src/components/TimeseriesDataExport/interfaces.ts

```diff
@@ -16,7 +16,7 @@ export type FetchCSVCall = (
 ) => Promise<string>;
 
 export type FetchTimeseriesCall = (
-  ids: InternalId[]
+  ids: IdEither[]
```

Comment:
I've just noticed that we have a problem here.
This type is exposed through `retrieveTimeseries` property, so we cannot actually break its signature. 
The only thing that comes to my mind is
```
export type FetchTimeseriesCall = (
ids?: number[]
externalIds?: string[]
```
bu I am not very satisfied with it

Reply:
and btw `number[] | string[]` solution will be breaking change as well in this case

Reply:
Seems to me that the current implementation won’t break anything: 
`export type FetchTimeseriesCall = (ids: IdEither[])...`
it will still work correctly when passing: 
`export type FetchTimeseriesCall = (ids: InternalId[]) ...`
I tested that. Or if you meant something else, please let me now.

But using type `FetchTimeseriesCall = (ids: number[], externalIds?: string[])...`
will definitely cause an error in `context!.timeseries.retrieve` method.

Reply:
what I am saying, is that if someone had

```
const fetch: FetchTimeseriesCall = (ids: InternalId[]):Promise<GetTimeSeriesMetadataDTO[]> => {
 //... their custom logic here
return Promise.resolve([]);
  }

  return (
      <TimeseriesDataExport {...props} retrieveTimeseries={fetch}/>
  );
```
their typescript would complain after update (although the code might remain functional).
@mrtraser is there a way to discriminate type inside the component?
so when only `timeseriesIds` provided, than `FetchTimeseriesCall` type will expect `InternalId` only?

...
PS: although it's a bit of breaking change, the chance someone used this exactly this way is pretty low, so we might just go for it

Reply:
Defining `FetchTimeseriesCall` type as:
```typescript
export type FetchTimeseriesCall = (
  ids: (IdEither | InternalId)[]
) => Promise<GetTimeSeriesMetadataDTO[]>;
```
should help here. But the solution far from a perfect one. So I would prefer to take a risk and go with `IdEither[]`, it doesn't break anything overall.

Reply:
It seems like we have to go for `IdEither []` after all. We'll mention the change in git commit description

Reply:
```typescript
export type FetchTimeseriesCall = (
  ids: (IdEither | InternalId)[]
) => Promise<GetTimeSeriesMetadataDTO[]>;
```
@polomani has checked solution and this will not help. The only one option here is to continue with `IdEither` I believe.